pkgs/development/libraries/qt-5/5.15/default.nix

```diff
@@ -180,6 +180,17 @@ let
       ./qtwebengine-darwin-no-platform-check.patch
       ./qtwebengine-mac-dont-set-dsymutil-path.patch
       ./qtwebengine-darwin-checks.patch
+    ] ++ lib.optionals stdenv.cc.isClang [
+      # 5.15.17: Fixes 'converts to incompatible function type [-Werror,-Wcast-function-type-strict]'
+      # in chromium harfbuzz dependency. This may be removed again if harfbuzz is updated
+      # to include the upstream fixes: https://github.com/harfbuzz/harfbuzz/commit/d88269c827895b38f99f7cf741fa60210d4d5169
+      # See https://trac.macports.org/ticket/70850
+      (fetchpatch {
+          name = "patch-qtwebengine_hb-ft.cc_error.patch";
+          url = "https://github.com/macports/macports-ports/commit/a9ba31ffe6bdd86b9c13737aa583e7c003f1fb58.patch";
+          sha256 = "sha256-8e5mi257MEDUrjYn1jb4VjHP8C3EYWXlNkkFTdpD+Oo=";
+          includes = [ "aqua/qt5/files/patch-qtwebengine_hb-ft.cc_error.diff" ];
+        })
```

Comment:
I don’t think this will work; it’ll just add a `.diff` file to the source tree.

Could you fetch the upstream HarfBuzz commit instead and use `extraPrefix`/`stripLen` to apply it to the right place in the WebEngine source?

Reply:
I see – in that case, we can probably just use `fetchurl` with `https://github.com/macports/macports-ports/raw/dd7bc40d8de48c762bf9757ce0a0672840c5d8c2/aqua/qt5/files/patch-qtwebengine_hb-ft.cc_error.diff`.

Reply:
@emilazy Fetchurl also does not work, see latest commit again. I don't know why.

Reply:
@emilazy I reverted back to using our own patch because fetchurl did not work due to timestamps.

Reply:
The timestamps are a normal part of the diff format. What’s the error you were getting?

Reply:
@emilazy I get the following error:
```console
applying patch /nix/store/3nny5pr2xq6cak6wyivia8ibcj1r29iq-patch-qtwebengine_hb-ft.cc_error.diff
can't find file to patch at input line 3
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- src/3rdparty/chromium/third_party/harfbuzz-ng/src/src/hb-ft.cc     2024-09-27 18:46:30
|+++ src/3rdparty/chromium/third_party/harfbuzz-ng/src/src/hb-ft.cc 2024-09-27 18:46:18
--------------------------
File to patch:
Skip this patch? [y]
Skipping patch.
4 out of 4 hunks ignored
```

Reply:
`src/3rdparty/chromium/third_party/harfbuzz-ng/src/src/hb-ft.cc` doesn’t look like the correct path (`src/src` seems unlikely) – it will probably work if you get the `extraPrefix` and `stripLen` values right.

Reply:
@emilazy I got confused and used `fetchurl` according to https://github.com/NixOS/nixpkgs/pull/358759#discussion_r1856171952 but now I use `fetchpatch` again with `extraPrefix = ""` and it finally works.